src/Kinesis.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Cascade\StatusStream;
+
+use Aws\Kinesis\KinesisClient;
+
+class Kinesis implements StatusStreamInterface
+{
+    private $component = '';
+    private $kinesis;
+    private $machineId;
+    private $streamName;
+    private $system = '';
+    private $subsystem = '';
+
+    public function __construct(KinesisClient $kinesis, $streamName)
+    {
+        $this->kinesis = $kinesis;
+        $this->streamName = $streamName;
+    }
+
+    /**
+     * @param string $system
+     * @param string $subsystem
+     * @param string $component
+     */
+    public function setSystemId($system, $subsystem = '', $component = '')
+    {
+        $this->system = $system;
+        $this->subsystem = $subsystem;
+        $this->component = $component;
+    }
+
+    /**
+     * @param array $machineId
+     */
+    public function setMachineId(array $machineId)
```

Comment:
Can it be a `mixed`, if it's a string, it's just a string. If it's an array it gets imploded with the ":" separators? Am I missing something about what a machineId is, and not understanding?